docs/pages/guides/tree-shaking.mdx

```diff
@@ -300,3 +302,211 @@ const { View, Image } = require('react-native-web');
+- Tree-shaking only runs in production bundles.
+- All modules in the Expo SDK are shipped as ESM and can be exhaustively tree-shaken.
+
+## Enabling tree shaking
+
+> Experimentally available in SDK 52 and greater
+
+<Step label="1">
+
+First, enable `experimentalImportSupport` and ensure your app builds and runs as expected. You may experience issues with require cycles or mixing commonjs and ESM imports.
+
+```js metro.config.js
+const { getDefaultConfig } = require('expo/metro-config');
+
+const config = getDefaultConfig(__dirname);
+
+config.transformer.getTransformOptions = async () => ({
+  transform: {
+    experimentalImportSupport: true,
+  },
+});
+
+module.exports = config;
+```
+
+Experimental import support uses Metro's version of the `@babel/plugin-transform-modules-commonjs` plugin. This drastically reduces the number of resolutions and simplifies your output bundle. This feature can be used with `inlineRequires` to further optimize your bundle experimentally.
+
+In the future, we'll enable this feature by default.
+
+</Step>
+
+<Step label="2">
+
+Toggle on the environment variable `EXPO_UNSTABLE_METRO_OPTIMIZE_GRAPH=1` to keep modules around until the entire graph is created. Ensure your app builds and runs as expected in production with this feature enabled before continuing.
+
+```sh .env
+EXPO_UNSTABLE_METRO_OPTIMIZE_GRAPH=1
+```
+
+This will only be used in production mode.
+
+</Step>
+
+<Step label="3">
+
+Toggle on the environment variable `EXPO_UNSTABLE_TREE_SHAKING=1` to enable the feature.
+
+```sh .env
+EXPO_UNSTABLE_TREE_SHAKING=1
+```
+
+This will only be used in production mode.
+
+</Step>
+
+<Step label="4">
+
+Bundle your app in production mode to see the effects of tree shaking.
+
+<Terminal cmd={['$ npx expo export']} />
+
+</Step>
+
+This feature is very experimental because it changes the fundamental structure of how Metro bundles code. By default Metro bundles everything on-demand and lazily to ensure the fastest possible development times. In contrast, tree shaking requires some transformation to be delayed until after the entire bundle has been created. This means less code can be cached, which is generally fine because tree shaking is a production-only feature and production bundles often don't use transform caches.
+
+## Barrel modules
+
+> Experimentally available in SDK 52 and greater
+
+With Expo tree shaking, star exports will automatically be expanded and shaken based on usage. For example, consider the following code snippet:
+
+```js Input
+export * from './icons';
+```
+
+The optimization pass will crawl `./icons` and add the exports to the current module. If the exports are unused, they will be removed from the production bundle.
+
+```js Expanded
+export { ArrowRight, ArrowLeft } from './icons';
+```
+
+This can then be shaken according to standard tree shaking rules. If you only import `ArrowRight`, then `ArrowLeft` will be removed from the production bundle.
```

Comment:
```suggestion
This will be shaken according to standard tree shaking rules. If you only import `ArrowRight`, then `ArrowLeft` will be removed from the production bundle.
```